doc/pyproject_toml.rst

```diff
@@ -67,12 +67,23 @@ requires
             "configparser; python_version == '2.7'"
       ]
 
+extras-require
+  Lists of packages needed for every optional feature. The requirements
+  are specified in the same format as for ``requires``. The requirements of
+  the two reserved extras ``test`` and ``doc`` as well as the extra ``dev``
+  are installed by ``flit install``. For example:
+
+  .. code-block:: toml
+
+      [tool.flit.metadata.extras-require]
+      test = ["pytest>=2.7.3", "pytest-cov"]
+      doc = ["sphinx"]
+
 dev-requires
-  Packages that are required for development. This field is in the same format
-  as ``requires``.
+  Packages that are required for development, other than the ones belonging to
+  the extras ``test`` and ``doc``. This field is in the same format as ``requires``.
 
-  These are not (yet) encoded in the wheel, but are used when doing
-  ``flit install``.
+  These are installed by ``flit install`` and encoded in the wheel as extra ``dev``.
```

Comment:
I would make this undocumented (or maybe leave it for now but mark it as deprecated). The canonical way will become `extras-require.dev`.